src/scripts/main.js

```diff
@@ -1,3 +1,226 @@
+      min: 18,
+      max: 90,
+    },
+  },
+  salary: {
+    label: 'Salary',
+    name: 'salary',
+    type: 'number',
+    'data-qa': 'salary',
+  },
+  startDate: '',
+};
+
+const strToNumber = (n) => Number(n.replace(/[^0-9.-]+/g, ''));
+
+// Add new employee form
+
+const createForm = (fields) => {
+  const formEl = document.createElement('form');
+
+  formEl.classList.add('new-employee-form');
+
+  for (const key in fields) {
+    if (fields[key]) {
+      const label = document.createElement('label');
+
+      label.textContent = `${fields[key].label}: `;
+
+      let field = document.createElement('input');
+
+      if (fields[key].hasOwnProperty('options')) {
+        field = document.createElement('select');
+
+        for (const option of fields[key].options) {
+          const optionEl = document.createElement('option');
+
+          optionEl.value = option;
+          optionEl.textContent = option;
+          field.appendChild(optionEl);
+        }
+      }
+
+      label.appendChild(field);
+
+      field.setAttribute('name', fields[key].name);
+      field.setAttribute('type', fields[key].type);
+      field.setAttribute('data-qa', fields[key]['data-qa']);
+      field.setAttribute('min', fields[key].validation?.min);
+      field.setAttribute('max', fields[key].validation?.max);
+      field.setAttribute('minlength', fields[key].validation?.minLength);
+      field.setAttribute('required', true);
+      formEl.appendChild(label);
+    }
+  }
+
+  return formEl;
+};
+
+const showNotification = (type, message) => {
+  const notification = document.createElement('div');
+
+  notification.innerHTML = `<h2 class='title'>${message}</h2>`;
+
+  notification.classList.add('notification', type);
+  notification.setAttribute('data-qa', 'notification');
+
+  body.appendChild(notification);
+
+  setTimeout(() => {
+    notification.remove();
+  }, 2000);
+};
+
+const createRow = (employee) => {
+  const row = document.createElement('tr');
+
+  for (const key in employee) {
+    const cell = document.createElement('td');
+
+    cell.textContent = employee[key];
+    row.appendChild(cell);
+  }
+
+  tbody.appendChild(row);
+};
+
+const submitHandler = (e) => {
+  if (!form.checkValidity()) {
+    showNotification('error', 'Error: Form is not valid');
+
+    return;
+  }
+
+  const formData = new FormData(form);
+  const employee = {};
+
+  for (const [key, value] of formData.entries()) {
+    if (key === 'salary') {
+      employee[key] = currencyFormat(value);
+      continue;
+    }
+    employee[key] = value;
+  }
+
+  createRow(employee);
+
+  showNotification('success', 'Employee added successfully');
+
+  init();
+
+  form.reset();
+};
+
+const createSubmitButton = () => {
+  const submit = document.createElement('button');
+
+  submit.setAttribute('type', 'submit');
+  submit.setAttribute('form', 'new-employee-form');
+  submit.addEventListener('click', submitHandler);
+
+  submit.textContent = 'Save to table';
+
+  return submit;
+};
```

Comment:
This function `createSubmitButton` is responsible for creating the submit button. It would be more efficient to separate this logic into smaller functions, for example, a function that creates a button, a function that sets the button attributes, and a function that adds an event listener to the button. This would make the code more maintainable and testable.